### DISCUSSION TITLE
rebuild the executable after code change

### DISCUSSION THREAD

**Discussion by SarahLu-NOAA** [ufs-community/ufs-srweather-app#1007] [Q&A] [answered]
Hello UFS / SRW experts,
I modified 3 routines in ccpp/data and ccpp/physics/physics and need to rebuild the ufs-weather-model executable.   I check user guide, FAQ, and repo discussions and can't find any info on re-building specific executable after code changes.
When I use './devbuild.sh --platform=<machine_name>', I can choose either [R]emove the existing directory, [C]ontinue building in the existing directory and [Q]uit this build script.    I try both R and C option.   I also check build directory for more information.   It's just not clear to me how to sort out why the build fails.  Thanks.

**Comment by mkavulich** [comment]
@SarahLu-NOAA The "Continue" option should work for re-building the SRW App after making source code changes, but this will build all executables (so you can not pick and choose if you have made multiple source code changes). The "Remove" option should also work but will remove the existing build directory and re-build everything from scratch, rather than re-using the existing build where possible.

The build logs for the CMake and Make step will be in `ufs-srweather-app/build/log.cmake` and `ufs-srweather-app/build/log.make`, so any errors encountered should be detailed in those files.

**Reply by mkavulich** [reply, marked as answer]
I will mention that the "Continue" option may not work as expected for changes in CCPP specifically, because the `ccpp_prebuild.py` script will not be re-run. It is probably best to re-compile the model entirely in this case.

**Reply by SarahLu-NOAA** [reply]
Thanks, @mkavulich 
Since I modified CCPP physics, I will use 'Remove' for a clean build.  I'll suggest including these info to FAQ, if possible.   Thanks.

**Reply by gspetro-NOAA** [reply]
Great idea! I'll look into drafting an FAQ. :)
